src/java/org/apache/cassandra/db/filter/RowFilter.java

```diff
@@ -136,7 +177,66 @@ public boolean hasExpressionOnClusteringOrRegularColumns()
         return false;
     }
 
-    protected abstract Transformation<BaseRowIterator<?>> filter(TableMetadata metadata, long nowInSec);
+    protected Transformation<BaseRowIterator<?>> filter(TableMetadata metadata, long nowInSec)
```

Comment:
It might be worth to comment here and/or in `isStrict()` that the multiple `filter` and `isSatisfiedBy` methods always use conjunction. Otherwise, one might wrongly understand that a non-strict filter would use disjunction in those methods.

Reply:
I think that what we'll see in [CASSANDRA-19007](https://issues.apache.org/jira/browse/CASSANDRA-19007) is this method becoming parameterized to handle the situation where we're hitting it from `ReadCommand#executeLocally()` (for normal, non-indexed AF queries). I can add a brief message here mentioning that 19007 will fix it, but that for now it ignores `isStrict()`?